helix-term/src/commands/typed.rs

```diff
@@ -1473,7 +1473,57 @@ fn tree_sitter_subtree(
             .root_node()
             .descendant_for_byte_range(from, to)
         {
-            let contents = format!("```tsq\n{}\n```", selected_node.to_sexp());
+            fn pretty_print_node(
+                node: Node<'_>,
+                sexpr: &mut String,
+                is_root: bool,
+                field_name: Option<&str>,
+                depth: usize,
+            ) -> anyhow::Result<()> {
+                fn is_visible(node: Node<'_>) -> bool {
+                    node.is_missing()
+                        || (node.is_named() && node.language().node_kind_is_visible(node.kind_id()))
+                }
+
+                if is_visible(node) {
+                    write!(sexpr, "{:depth$}", "")?;
+
+                    if let Some(field_name) = field_name {
+                        write!(sexpr, "{}: ", field_name)?;
+                    }
+
+                    write!(sexpr, "({}", node.kind())?;
+                } else if is_root {
+                    write!(sexpr, "(\"{}\")", node.kind())?;
+                }
+
+                for child_idx in 0..node.child_count() {
+                    if let Some(child) = node.child(child_idx) {
+                        if is_visible(child) {
+                            sexpr.push('\n');
+                        }
+
+                        pretty_print_node(
+                            child,
+                            sexpr,
+                            false,
+                            node.field_name_for_child(child_idx as u32),
+                            depth + 2,
+                        )?;
+                    }
+                }
+
+                if is_visible(node) {
+                    write!(sexpr, ")")?;
+                }
+
+                Ok(())
+            }
+
+            let mut sexpr = String::new();
+            pretty_print_node(selected_node, &mut sexpr, true, None, 0)?;
+
+            let contents = format!("```tsq\n{}\n```", sexpr);
```

Comment:
This will reallocate a string only to over `sexpr`, you could start `sexpr` with 

```rust
let mut sexpr = String::from("```tsq\n");
pretty_print_node(selected_node, &mut sexpr, true, None, 0)?;
sexpr.push_str("\n```");
```